fenix/app/src/androidTest/java/org/mozilla/fenix/ui/AddToHomeScreenTest.kt

```diff
@@ -0,0 +1,118 @@
+import org.mozilla.fenix.helpers.TestAssetHelper
+import org.mozilla.fenix.helpers.TestHelper
+import org.mozilla.fenix.ui.robots.browserScreen
+import org.mozilla.fenix.ui.robots.homeScreen
+import org.mozilla.fenix.ui.robots.navigationToolbar
+import org.mozilla.fenix.ui.robots.searchScreen
+
+class AddToHomeScreenTest {
+    private lateinit var mockWebServer: MockWebServer
+    private val downloadTestPage =
+        "https://storage.googleapis.com/mobile_test_assets/test_app/downloads.html"
+    private val pdfFileName = "washington.pdf"
+    private val pdfFileURL = "storage.googleapis.com/mobile_test_assets/public/washington.pdf"
+    private val pdfFileContent = "Washington Crossing the Delaware"
+
+    @get:Rule
+    val activityIntentTestRule = HomeActivityIntentTestRule.withDefaultSettingsOverrides()
+
+    @Before
+    fun setUp() {
+        mockWebServer = MockWebServer().apply {
+            dispatcher = AndroidAssetDispatcher()
+            start()
+        }
+    }
+
+    @After
+    fun tearDown() {
+        mockWebServer.shutdown()
+    }
+
+    // Verifies the Add to home screen option in a tab's 3 dot menu
+    @Test
+    fun mainMenuAddToHomeScreenTest() {
+        val website = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+        val shortcutTitle = TestHelper.generateRandomString(5)
+
+        homeScreen {
+        }.openNavigationToolbar {
+        }.enterURLAndEnterToBrowser(website.url) {
+        }.openThreeDotMenu {
+            expandMenu()
+        }.openAddToHomeScreen {
+            clickCancelShortcutButton()
+        }
+
+        browserScreen {
+        }.openThreeDotMenu {
+            expandMenu()
+        }.openAddToHomeScreen {
+            verifyShortcutTextFieldTitle("Test_Page_1")
+            addShortcutName(shortcutTitle)
+            clickAddShortcutButton()
+            clickAddAutomaticallyButton()
+        }.openHomeScreenShortcut(shortcutTitle) {
+            verifyUrl(website.url.toString())
+            verifyTabCounter("1")
+        }
+    }
+
+    @Test
+    fun addPrivateBrowsingShortcutTest() {
+        homeScreen {
+        }.dismissOnboarding()
+
+        homeScreen {
+        }.triggerPrivateBrowsingShortcutPrompt {
+            verifyNoThanksPrivateBrowsingShortcutButton()
+            verifyAddPrivateBrowsingShortcutButton()
+            clickAddPrivateBrowsingShortcutButton()
+            clickAddAutomaticallyButton()
+        }.openHomeScreenShortcut("Private ${TestHelper.appName}") {}
+        searchScreen {
+            verifySearchView()
+        }.dismissSearchBar {
+            verifyPrivateSessionMessage()
+        }
+    }
+
+    @SmokeTest
+    @Test
+    fun addPDFToHomeScreenTest() {
+        TestHelper.runWithCondition(
+            // Returns the GeckoView channel set for the current version, if a feature is limited to Nightly or Beta.
+            // Once this feature lands in Beta/RC we should remove the wrapper.
```

Comment:
From the condition used, I understand the feature is already in Beta, so the comment should reflect that it needs to be enabled just in Release before removing the wrapper.

Reply:
Done, thanks !